Trendyol.App.WebApi/Filters/BasicAuthenticationFilter.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Principal;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Filters;
+
+namespace Trendyol.App.WebApi.Filters
+{
+    internal class BasicAuthenticationFilter : IAuthenticationFilter
+    {
+        private readonly IAuthenticationChecker _checker;
+
+        public BasicAuthenticationFilter(IAuthenticationChecker checker)
+        {
+            _checker = checker;
+        }
+
+        public bool AllowMultiple => true;
+
+        public async Task AuthenticateAsync(HttpAuthenticationContext context, CancellationToken cancellationToken)
+        {
+            HttpRequestMessage request = context.Request;
+            AuthenticationHeaderValue authorization = request.Headers.Authorization;
+
+            if (authorization == null)
+            {
+                return;
+            }
+
+            if (authorization.Scheme != "Basic")
+            {
+                return;
+            }
+
+            if (string.IsNullOrEmpty(authorization.Parameter))
+            {
+                context.ErrorResult = new AuthenticationFailureResult("Missing credentials", request);
+                return;
+            }
+
+            Tuple<string, string> userNameAndPasword = ExtractUserNameAndPassword(authorization.Parameter);
+            if (userNameAndPasword == null)
+            {
+                context.ErrorResult = new AuthenticationFailureResult("Invalid credentials", request);
+                return;
+            }
+
+            string userName = userNameAndPasword.Item1;
+            string password = userNameAndPasword.Item2;
+
+            IPrincipal principal = null;
+            if (_checker.Check(userName, password, cancellationToken))
+            {
+                GenericIdentity identity = new GenericIdentity(userName);
+                principal = new GenericPrincipal(identity, new List<string>().ToArray());
```

Comment:
We need to create a ClaimsPrincipal here with authenticated user's claims.
That means you also need to have some dependency that could provide you relevant user information via username, such as claims.